### PR TITLE
expose auth service

### PR DESCRIPTION
**Pull request by aleoli**
# Description

Provide a way to contact the auth service form outside of the cluster

## What's new

* add the auth service to helm chart
  * add a NodePort service
  * set its port in mDNS discovery
* provide a TLS certificate to the service with an initContainer

Ref. issue #371

### CODE REVIEW

File: internal/discovery/authData.go [hunk 1]
```diff
@@ -59,9 +77,10 @@ func getReachable(ips []net.IP, port int) (*net.IP, error) {
 	}
 }
 
-func isReachable(address string, port int) bool {
-	//_, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%d", address, port), 500*time.Millisecond)
-	//return err == nil
-	// TODO: no service is available yet
-	return true
+// check if this address + port is reachable with TCP
+// the service is reachable is we are able to establish a TCP connection before a
```
Comment (palexster) [inline internal/discovery/authData.go:81]: before a?

Comment (aleoli) [reply internal/discovery/authData.go:81]: I'm sorry... It was ".. before the timeout"

File: internal/discovery/foreign.go [hunk 2]
```diff
@@ -133,36 +143,51 @@ func (discovery *DiscoveryCtrl) createForeign(data *discoveryData, sd *v1alpha1.
...
+			fc.Status.Outgoing.CaDataRef = nil
 		}
-		fc.Status.Outgoing.CaDataRef = nil
 		fc.LastUpdateNow()
 		tmp, err := discovery.crdClient.Resource("foreignclusters").Update(fc.Name, fc, metav1.UpdateOptions{})
 		if err != nil {
 			klog.Error(err)
 			return nil, false, err
 		}
+		klog.V(4).Info("TTL updated")
```
Comment (mlavacca) [inline internal/discovery/foreign.go:181]: Isn't this log redundant with the one in line188?

Comment (aleoli) [reply internal/discovery/foreign.go]: yes, thanks, I removed the second one

**Review by mlavacca — APPROVED** [review]

**Review by palexster — APPROVED** [review]

**Pull request merged**